File: services/fraud-detection-service/src/main/java/com/zerofake/fraud/service/FraudReportService.java
package com.zerofake.fraud.service;

import com.zerofake.fraud.dto.request.FraudReportRequest;
import com.zerofake.fraud.dto.response.FraudReportResponse;

import java.util.List;
import java.util.UUID;

public interface FraudReportService {

    FraudReportResponse createFraudReport(FraudReportRequest request);

    FraudReportResponse getFraudReportById(UUID reportId);

    List<FraudReportResponse> getAllFraudReports();

}

File: zerofake-frontend/src/services/fraudReportService.ts
import { fraudApi } from "@/api/fraudApi";

import type {
  FraudReportRequest,
  FraudReportResponse,
} from "@/types/fraud";

class FraudReportService {
  async getFraudReports(): Promise<FraudReportResponse[]> {
    const response = await fraudApi.getFraudReports();
    return response.data ?? [];
  }

  async createFraudReport(
    request: FraudReportRequest
  ): Promise<FraudReportResponse> {
    const response = await fraudApi.createFraudReport(request);
    return response.data;
  }
}

export const fraudReportService = new FraudReportService();
